apps/src/studio/rocketHeightLogic.js

```diff
@@ -0,0 +1,51 @@
+var CustomGameLogic = require('./customGameLogic');
+var studioConstants = require('./constants');
+var Direction = studioConstants.Direction;
+var codegen = require('../codegen');
+var api = require('./api');
+
+/**
+ * Custom logic for the Rocket Height levels
+ * @constructor
+ * @implements CustomGameLogic
+ */
+var RocketHeightLogic = function (studio) {
+  CustomGameLogic.apply(this, arguments);
+  this.rocketIndex = 0;
+  this.rocket = null;
+  this.last = Date.now();
+  this.seconds = 0;
+  this.height = 0;
+};
+RocketHeightLogic.inherits(CustomGameLogic);
+
+RocketHeightLogic.prototype.onTick = function () {
+  
+  // Update the rocket once a second
+  if (Date.now() - this.last < 1000) {
+    return;
+  } else {
+    this.last = Date.now();
+    this.seconds++;
+  }
+  
+  this.rocket = this.studio_.sprite[this.rocketIndex];
+  
+  // Display the rocket height and time elapsed
+  this.height = this.rocket_height(this.seconds);
+  this.rocket.y = this.studio_.MAZE_HEIGHT - (this.height + this.rocket.height);
+  this.rocket.dir = Direction.NONE;
+  this.studio_.scoreText = 'Time: ' + this.seconds + ' | Height: ' + this.height;
+  this.studio_.displayScore();
```

Comment:
I made them members in case I end up needing to access them for success/failure checking.